modules/core/arrow-test/src/main/kotlin/arrow/test/UnitSpec.kt

```diff
@@ -11,28 +10,19 @@ import io.kotlintest.specs.AbstractStringSpec
  */
 abstract class UnitSpec : AbstractStringSpec() {
 
+  private val lawTestCases = mutableListOf<TestCase>()
+
   fun testLaws(vararg laws: List<Law>): List<TestCase> {
-   laws
+    return laws
      .flatMap { list: List<Law> -> list.asIterable() }
      .distinctBy { law: Law -> law.name }
      .map { law: Law ->
-       createTestCase(law.name, law.test, defaultTestCaseConfig, TestType.Test)
+       val lawTestCase = createTestCase(law.name, law.test, defaultTestCaseConfig, TestType.Test)
```

Comment:
Shouldn't we use `addTestCase` here?

Reply:
the add test case requires unique names and the test laws are duplicated in some many places

Reply:
We're missing tests this way, right? I.e.

`FunctorLaws.laws(IO.functor())` & `TraverseLaws.laws(IO.traverse())` the functor laws aren't tested for `Traverse<ForIO>` but the functor logic might not be defined by the same functor instance.

Reply:
For those cases, you can have multiple testLaws declarations and kotlintest will execute them without errors even with duplicated names, the best example is the [FlowableKTests](https://github.com/arrow-kt/arrow/blob/arf-solve-update-kotlintest-problems/modules/effects/arrow-effects-rx2-data/src/test/kotlin/arrow/effects/FlowableKTests.kt) where we execute multiple times the same laws with different parameters. This works now because, only for law test, I deleted the unique name restriction